### PR TITLE
[LogisticRegression] Support sparse matrix in transform

### PR DESCRIPTION
**Pull request by lijinf2**

### CODE REVIEW

**Comment by lijinf2** [comment]
build

File: python/src/spark_rapids_ml/core.py [hunk 1]
```diff
@@ -1436,7 +1482,9 @@ def predict_udf(iterator: Iterator[pd.DataFrame]) -> Iterator[pd.Series]:
                 cuml_objects[0] if isinstance(cuml_objects, list) else cuml_objects
             )
             for pdf in iterator:
-                if not input_is_multi_cols:
+                if use_sparse_array:
```
Comment (eordentlich) [inline python/src/spark_rapids_ml/core.py:1485]: Looks good.   I think you just need to make this change also where transform evaluate invokes cuml and should be good.  Here I think https://github.com/NVIDIA/spark-rapids-ml/blob/branch-23.12/python/src/spark_rapids_ml/core.py#L1298

Comment (eordentlich) [reply python/src/spark_rapids_ml/core.py:1485]: And add some tests.

**Comment by lijinf2** [comment]
build

File: python/src/spark_rapids_ml/core.py (hunk 1 above)
Comment (lijinf2) [reply python/src/spark_rapids_ml/core.py:1492]: Yeah, revised the code and tested through cross validator on sparse datasets.

**Review by eordentlich — APPROVED** [review]
👍

**Comment by eordentlich** [comment]
Would be great to get an example notebook for the nlp example dataset from the test.

**Pull request merged**